content/references/rippled-api/api-conventions/serialization.md

```diff
@@ -0,0 +1,285 @@
+# Serialization Format
+[[Source]<br>](https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/STObject.cpp#L696-L718 "Source")
+
+This page describes the XRP Ledger's canonical binary format for transactions and other data. This binary format is necessary to create and verify digital signatures of those transactions' contents, and is also used in other places. The [rippled APIs](rippled-api.html) typically use JSON to communicate with client applications. However, JSON is unsuitable as a format for serializing transactions for being digitally signed, because JSON can represent the same data in many different but equivalent ways.
+
+The process of serializing a transaction from JSON or any other representation into their canonical binary format can be summarized with these steps:
+
+1. Make sure all required fields are provided, including any required but "auto-fillable" fields.
+
+    **Note:** The `SigningPubKey` must also be provided at this step. When signing, you can derive this key from the secret key that is provided for signing.
+
+2. Convert each field's data into its "internal" binary format.
+3. Sort the fields in canonical order.
+4. Prefix each field with a Field ID.
+5. Concatenate the fields (including prefixes) in their sorted order.
+
```

Comment:
Would it be helpful to anchor link these steps to the sections below that describe how to perform them? For step 2, link "internal binary format" to "Internal Format," for step 3 link "canonical order" to "Canonical Field Order," and for step 4 link "Field ID" to "Field IDs."  Just to give the reader a heads up that further guidance on these steps is coming up?

Reply:
Great idea! Added links to all the steps except the concatenate one (concatenating things is trivial and not unique to the XRP Ledger). That also reminded me to move a couple sections around so that some things that aren't part of the canonical ordering stuff aren't under the "Canonical Field Order" header.